File: MODULO 2/06) DOWNLOAD IMAGENS/IMAGENS.py
import telebot
import requests
import os

TOKEN = "SEU_TOKEN"
bot = telebot.TeleBot(TOKEN)

UNSPLASH_ACCESS_KEY = "SUA_CHAVE_DO_UNSPLASH"
UNSPLASH_API_URL = "https://api.unsplash.com/photos/random"

@bot.message_handler(commands=["start"])
def start(message):
    bot.send_message(message.chat.id, "Olá! Envie uma descrição de uma imagem ou uma palavra-chave.")

@bot.message_handler(func=lambda message: True)
def search_image(message):
    query = message.text
    headers = {
        "Accept-Version": "v1",
        "Authorization": f"Client-ID {UNSPLASH_ACCESS_KEY}"
    }
    params = {
        "query": query,
        "orientation": "landscape"
    }

    response = requests.get(UNSPLASH_API_URL, headers=headers, params=params)
    data = response.json()

    if response.status_code == 200 and data.get("urls"):
        image_url = data["urls"]["regular"]
        bot.send_photo(message.chat.id, image_url)
    else:
        bot.send_message(message.chat.id, "Não foi possível encontrar uma imagem correspondente.")

bot.polling()

File: MODULO 2/06) DOWNLOAD IMAGENS/IMAGENS.js
const TelegramBot = require('node-telegram-bot-api');
const axios = require('axios');

const TOKEN = 'SEU_TOKEN';
const bot = new TelegramBot(TOKEN, { polling: true });

const PEXELS_API_KEY = 'SUA_CHAVE_DO_PEXELS';

bot.onText(/\/start/, (msg) => {
  const chatId = msg.chat.id;
  bot.sendMessage(chatId, 'Olá! Envie uma descrição de uma imagem ou uma palavra-chave.');
});

bot.onText(/.*/, async (msg, match) => {
  const chatId = msg.chat.id;
  const query = match[0];
  const pexelsApiUrl = `https://api.pexels.com/v1/search?query=${encodeURIComponent(query)}&per_page=1`;
  
  try {
    const response = await axios.get(pexelsApiUrl, {
      headers: {
        Authorization: PEXELS_API_KEY,
      },
    });

    if (response.status === 200 && response.data.photos.length > 0) {
      const imageUrl = response.data.photos[0].src.original;
      bot.sendPhoto(chatId, imageUrl);
    } else {
      bot.sendMessage(chatId, 'Não foi possível encontrar uma imagem correspondente.');
    }
  } catch (error) {
    bot.sendMessage(chatId, 'Ocorreu um erro ao buscar a imagem.');
  }
});
